packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp

```diff
@@ -5647,6 +5657,122 @@ namespace Tpetra {
     }
   }
 
+  template <class LocalOrdinal, class GlobalOrdinal, class Node>
+  void
+  CrsGraph<LocalOrdinal, GlobalOrdinal, Node>::
+  resizeForIncomingDataImpl(const Kokkos::UnorderedMap<LocalOrdinal, size_t>& padding)
+  {
+    const char tfecfFuncName[] = "resizeForIncomingDataImpl";
+    using device_type = typename Node::device_type;
+    using execution_space = typename device_type::execution_space;
+    using row_ptrs_type = typename local_graph_type::row_map_type::non_const_type;
+    using indices_type = t_GlobalOrdinal_1D;
+    using range_policy = Kokkos::RangePolicy<execution_space, Kokkos::IndexType<LocalOrdinal>>;
+    using Tpetra::Details::resizeRowPtrsAndIndices;
+
+    if (! this->indicesAreAllocated()) {
+      allocateIndices(GlobalIndices);
+    }
+    TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC(
+      ! this->isGloballyIndexed(), std::runtime_error,
+      ": must be globally indexed to resize!\n");
+
+    // Making copies here because k_rowPtrs_ has a const type. Otherwise, we
+    // would use it directly.
+    indices_type indices("indices", this->k_gblInds1D_.extent(0));
+    Kokkos::deep_copy(indices, this->k_gblInds1D_);
+
+    row_ptrs_type row_ptrs_beg("row_ptrs_beg", this->k_rowPtrs_.extent(0));
+    Kokkos::deep_copy(row_ptrs_beg, this->k_rowPtrs_);
+
+    const auto N = (row_ptrs_beg.extent(0) == 0 ? 0 : row_ptrs_beg.extent(0) - 1);
```

Comment:
The two branches of the ternary operator may have different types here.  I would recommend explicitly casting each to e.g., `size_t`.

Reply:
Doh!  Good catch!!